core/src/main/java/org/wildfly/channel/Repository.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.channel;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Objects;
+
+public class Repository {
+    private String id;
+    private String url;
+
+    @JsonCreator
+    public Repository(@JsonProperty(value = "id") String id, @JsonProperty(value = "url") String url) {
```

Comment:
is the `id` required?

Reply:
I'm not sure. I guess we could auto-generate the id if it's not provided. But I think it might possibly cause some maven cache issue if the ids collide

Reply:
In the Maven world repository `id` is used to track the state in the local cache. So if you allow resolution to go through a Maven cache the `id`s must match.